pkg/cmd/operator/analyze.go

```diff
@@ -101,12 +107,40 @@ func (o *AnalyzeOptions) Complete() error {
 		return err
 	}
 
+	o.kubeClient, err = kubernetes.NewForConfig(o.ProtoConfig)
+	if err != nil {
+		return fmt.Errorf("can't build kubernetes clientset: %w", err)
+	}
+	o.scyllaClient, err = scyllaversioned.NewForConfig(o.RestConfig)
+	if err != nil {
+		return fmt.Errorf("can't build scylla clientset: %w", err)
+	}
+
 	return nil
 }
 
 func (o *AnalyzeOptions) Run(streams genericclioptions.IOStreams, cmd *cobra.Command) error {
 	klog.Infof("%s version %s", cmd.Name(), version.Get())
 	cliflag.PrintFlags(cmd.Flags())
 
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var (
+		_   analyze.DataSource
+		err error
+	)
+	if len(o.ArchivePath) > 0 {
+		_, err = analyze.DataSource{}, errors.New("must-gather archives are currently unsupported")
+	} else {
+		_, err = analyze.NewDataSourceFromClients(ctx, o.kubeClient, o.scyllaClient)
+		if err != nil {
+			err = fmt.Errorf("can't build data source from clients: %w", err)
+		}
+	}
+
+	if err != nil {
+		return err
+	}
```

Comment:
nit: you could just return early in both branches

Reply:
done